### ISSUE TITLE
Performance Issues

### ISSUE THREAD

**Issue by ssambhi900** [th3jesta/ha-lcars#82]
Is there anything that can be done about load times for this theme?  Every time we switch pages, each item takes about 1 second to render, and they do it one at a time.  When a light switch is clicked, the styling disappears for 0.5 seconds, and comes back, making it look "broken".  Running it on a Fire 10 HD.  Any other theme doesn't experience this issue.

Love this theme, just want it to perform as quick and smooth as other themes.  If anyone has suggestions, that's good too!

Thx!!

**Comment by th3jesta** [comment]
I would start with disabling the texture using the entity you were instructed to create. It will look flatter (seems many people prefer this) and it will reduce the number items that need to load. The textures are a bit heavy. 
Beyond that, you could try putting Cloudflare (free) in front of your site if it's exposed to the internet externally. It will cache some of the assets which should improve load times, especially if the system you are hosting it on is lower power. Ultimately, though, it comes down to the hardware that hosts Home Assistant, the network it lives in, and the hardware that displays it. By necessity, this theme is heavier than most other themes.

**Comment by ssambhi900** [comment]
Thanks. Can you clarify what you mean by texture? We’re running it all locally. It works fine when accessing in my iPhone. It’s just locally in the Fire tablets where it’s slow to render. Might just be the complexity of the theme as we don’t have issues with other themes.

**Comment by th3jesta** [comment]
Yes, unfortunately the theme is heavy and adds a lot of extra CSS. The only other thing I would suggest is simplifying the dashboards that load on the problematic devices.
